document-store/src/main/java/org/hypertrace/core/documentstore/expression/ArithmeticOperator.java

```diff
@@ -0,0 +1,15 @@
+package org.hypertrace.core.documentstore.expression;
+
+public enum ArithmeticOperator {
+  // Unary operations
+  ABS,
+  FLOOR,
+
+  // Binary operations
+  SUBTRACT, // Minuend and then subtrahend
```

Comment:
both of these could be considered n-ary in the same way as add and multiply. e.g. `A - B - C` is just as valid as `A + B + C`